buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy

```diff
@@ -0,0 +1,747 @@
+        props.each { pName, o ->
+            log.info("\t processing property {}: {}", pName, o)
+            Map pDesc
+            if (o instanceof Map) {
+                pDesc = o as Map
+            } else {
+                throw new GradleException("prop desc not map: " + o.class.name)
+            }
+            log.info("\t processing property {}: {}", pName, pDesc)
+
+            // Properties are required if there is no default...
+            Map<String, Object> propDesc = [ required: pDesc.default == null ]
+            // Here, we need to be wary of Groovy Truth. Check for non-null since default values of '', false, etc.
+            // are "false" in Groovy Truth land.
+            if (pDesc.default != null) {
+                propDesc.default = pDesc.default
+            }
+            if (pDesc.description != null) {
+                propDesc.description = pDesc.description
+            } else if (pDesc.title != null) {
+                propDesc.description = pDesc.title
+            } else {
+                propDesc.description = 'Value for ' + pDesc.pName + '.'
+            }
+            if (pDesc.type != null) {
+                propDesc.type = StringUtils.capitalize(pDesc.type as String)
+            } else {
+                propDesc.type = 'String'    // FIXME -- Better Default?
+            }
+            if (PASSWORD_LIKE.contains(pDesc.format?.toLowerCase()) ||
+                    ((String) pDesc[PROPERTY_X_DESCRIPTORS])?.contains(DISPLAY_HIDE_VALUE)) {
+                // for things that are passwords or are marked to be masked in display, we will require a secret.
+                propDesc.type = 'Secret'
+                propDesc.description += ' (Please provide the name of the Vantiq Secret containing ' + 'this value.)'
+            }
+            if (pDesc.enum != null) {
+                propDesc.enum = pDesc.enum
+            }
+            log.info('PropDesc for {} : {}', pName, propDesc)
+            cfgProps[pName] = propDesc
+            // FIXME -- handle secrets if appropriate
+            cfgMappings[pName] = [resource: VANTIQ_SOURCES, resourceId: sourceName,
+                                  property:
+                                      "config.${CONFIG_CAMEL_RUNTIME}.${CAMEL_RUNTIME_PROPERTY_VALUES}.${pName}"] as
+                                            Map<String, Object>
+        }
+        project.configurationMappings = cfgMappings
+        project.configurationProperties = cfgProps
+        def projectJson =  JsonOutput.prettyPrint(JsonOutput.toJson(project))
+        log.info('Built project {}:\n {}', project.name, projectJson)
+        Map<String, Object> retVal = [:]
+
+        retVal.path = writeVantiqEntity(VANTIQ_PROJECTS, kameletAssemblyDir, packageName, projectName + JSON_SUFFIX,
+            projectJson, false)
+        retVal.reference = buildResourceRef(VANTIQ_PROJECTS, packageName, projectName)
+        retVal.vailName = packageName
+        return retVal
+    }
+
```

Comment:
points below this are just there due to project & class rename.  No substantive changes.